src/dune_rules/coq/coq_lib.ml

```diff
@@ -504,16 +505,30 @@ module DB = struct
     let create_from_stanza coq_db db dir stanza =
       Memo.exec memo (coq_db, db, dir, stanza)
 
-    (* XXX: Memoize? This is pretty cheap so not sure worth the cost *)
+    let directory_of_file p =
+      Path.of_string (Path.to_string p |> Filename.dirname)
+
+    (* XXX: Memoize? This is pretty cheap so not sure worth the cost,
+       still called too much I have observed, suspicious! *)
     let create_from_coqpath ~boot_id cp =
       let name = Coq_path.name cp in
       let installed_root = Coq_path.path cp in
       let implicit = Coq_path.stdlib cp in
       let cmxs = Coq_path.cmxs cp in
+      let cmxs_directories =
+        Path.Set.of_list_map ~f:directory_of_file cmxs |> Path.Set.to_list
```

Comment:
Better way to do this?

Reply:
Almost done in my tree.

Reply:
If the goal is to remove duplicates, this is the way I have seen it being done elsewhere. Alreternatively we could just always use a map rather than a list.

Reply:
Actually accumulating the dirs directly makes the paths unique (and removes the need for parent, so I will push that trivial change)